ciao-controller/api/api.go

```diff
@@ -0,0 +1,516 @@
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/01org/ciao/ciao-controller/types"
+	"github.com/golang/glog"
+	"github.com/gorilla/mux"
+)
+
+// Port is the default port number for the ciao API.
+const Port = 8889
```

Comment:
@rbradford This is just the default value. The caller who starts the service can choose not to use it.